src/Console/ConsoleIo.php

```diff
@@ -176,7 +176,7 @@ public function quiet($message, int $newlines = 1)
      * @param string|array $message A string or an array of strings to output
      * @param int $newlines Number of newlines to append
      * @param int $level The message's output level, see above.
-     * @return int|bool The number of bytes returned from writing to stdout.
+     * @return int|true The number of bytes returned from writing to stdout.
```

Comment:
Returning `true` for `ConsoleIo::out()` holds no value. If the passed level is greater than current level and no output is written it should just return `null`.

Other option is to remove the `int` casting in statement `$this->_lastWritten = (int)$this->_out->write($message, $newlines);` and revert the return type to `int|bool`, since the chain of methods calls to write output ends with `fwrite()` which can return `false` on failure.

Reply:
I know : ) I agree with u. Could also be int 0 here.